### DISCUSSION TITLE
Error: SEARCH-3-1-11-v19.12 FIND EVENTS - FORWARD AND BACKWARD SEARCH (SEARCH ENDPOINTS INSIDE RECORDING ENDPOINTS) is failing with error no virtual event found.

### DISCUSSION THREAD

**Discussion by sagarmodasiya** [onvif/testspecs#56] [Q&A] [answered]
Hi Team,

We are getting the below mentioned error while running onvif conformance test with our product.

Error:
SEARCH-3-1-11-v19.12 FIND EVENTS - FORWARD AND BACKWARD SEARCH (SEARCH ENDPOINTS INSIDE RECORDING ENDPOINTS) is failing with error no virtual event found.

We referred to ONVIF documents related to this test case but didn't get much information on virtual events.
Can you please provide some additional details for the same? We kept all configurations the same as mentioned in pre-requisition of the ONVIF test.

Here I have attached a detailed report of error in the xml file, please let us know your inputs.
![IMG-8720](https://github.com/onvif/testspecs/assets/143701500/5332ddf8-d3ee-4b35-a516-3806434f31f3)

[Onvif_test_failed_report.pdf](https://github.com/onvif/testspecs/files/12484406/Onvif_test_failed_report.pdf)

**Comment by madhurao68** [comment]
Hello!

According to the logs following is the search request from DTT:
- StartPoint = 2023-08-28T15:50:28Z
- EndPoint = 2023-08-28T16:18:20Z
- IncludeStartState = true

And all results are events dated with Time = 2023-08-28T15:50:00Z and with StartStateEvent = false. So first of all events are out of the search interval, and second point there are no events with StartStateEvent = true.

Virtual start state events allow Client to determine the state of the track at Start Point of search. Please find the example for tns1:RecordingHistory/Track/State event below:

![first](https://github.com/onvif/testspecs/assets/73811069/c6139e3f-36e6-4f1d-a925-35c1354e79b7)

**Reply by sagarmodasiya** [reply]
Hello, 

Thank for the your response. But we are still facing the issue with another step in same test case.
Details are as follows:

ERROR:

<Number>49</Number> : <StepName>Check that virtual events with topic tns1:RecordingHistory/Recording/State are received at end point</StepName>

            <Message>No virtual events with topic tns1:RecordingHistory/Recording/State are received at end point</Message>

Below are the mentioned details of startpoint, endpoint and includestartstate flag as per the onvif request. Here, I have attached xml file for your reference. Please verify in step 30 Find Events.

StartPoint : 2023-09-13T19:04:30Z
EndPoint : 2023-09-13T19:00:29Z
IncludeStartState : true

We were getting Get Events Search results for the endpoint which is same as endpoint requested in Find Event step. Still unable to find event. Please refer Get Events Search results response in xml file.

          &lt;tt:RecordingToken&gt;OnvifRecordingToken&lt;/tt:RecordingToken&gt;
          &lt;tt:TrackToken&gt;VideoTrackToken&lt;/tt:TrackToken&gt;
          &lt;tt:Time&gt;2023-09-13T19:00:29Z&lt;/tt:Time&gt;

![j8gxrc](https://github.com/onvif/testspecs/assets/143701500/549e6173-8462-42a5-8b2f-86d44b3b4d12)

XML File: 
[ONVIF_Response_XML.pdf](https://github.com/onvif/testspecs/files/12605297/ONVIF_Response_XML.pdf)

**Comment by madhurao68** [comment, marked as answer]
Unfortunately, attached file has only test log of the tests, so we could not analyze exact massages.
But according to error most probably virtual events for end point of search is missing.

The failed test steps are for backward search, and for backward search device shall return virtual events for both StartPoint and EndPoint time marks.

Here is the picture:
![img1](https://github.com/onvif/testspecs/assets/73811069/0ae33d47-d9bf-40aa-82b0-ab214490fb12)
